eng/versioning/version_client.txt

```diff
@@ -87,12 +87,12 @@ com.azure:azure-cosmos-dotnet-benchmark;4.0.1-beta.1;4.0.1-beta.1
 com.azure.cosmos.spark:azure-cosmos-spark_3_2-12;1.0.0-beta.1;1.0.0-beta.1
 com.azure.cosmos.spark:azure-cosmos-spark_3-1_2-12;4.8.0;4.9.0-beta.1
 com.azure.cosmos.spark:azure-cosmos-spark_3-2_2-12;4.8.0;4.9.0-beta.1
-com.azure:azure-cosmos-encryption;1.0.0;1.1.0-beta.1
+com.azure:azure-cosmos-encryption;1.0.1;1.1.0-beta.1
 com.azure:azure-data-appconfiguration;1.3.2;1.4.0-beta.1
 com.azure:azure-data-appconfiguration-perf;1.0.0-beta.1;1.0.0-beta.1
 com.azure:azure-data-schemaregistry;1.1.1;1.2.0-beta.1
 com.azure:azure-data-schemaregistry-apacheavro;1.0.0-beta.11;1.0.0-beta.12
-com.azure:azure-data-tables;12.2.1;12.3.0-beta.1
+com.azure:azure-data-tables;12.2.2;12.3.0-beta.1
```

Comment:
Were these not updated because the auto-increment PRs were not merged? We should have a check to ensure all autoincrement PRs are merged/closed and not left open leading to stale versions in this version_client.txt.

Reply:
Yes - I now have scripts that run as part of the bomgeneration to use maven as our source of truth and update our system accordingly if anything was missed. I do not want to depend on PR data and rely on bot correctly creating PRs. 

My next aim is to create the correct state for BOM before patching even starts so mismatches won't cause wrong patches, I am nearly finished with that, may need a few more cycles.

Reply:
We should ensure that we are moving the version forward if we rely on Maven central. If the version_client.txt has a newer version than what is found on Maven central, then it might be due to delay in updating Maven cache and we should assume that version_client is updated.